orthoinference/src/main/java/org/reactome/orthoinference/ReactionInferrer.java

```diff
@@ -192,35 +209,50 @@ private static boolean inferReactionCatalysts(GKInstance reactionInst, GKInstanc
 				infCatalystInst.addAttributeValue(_displayName, catalystInst.getAttributeValue(_displayName));
 				infCatalystInst = InstanceUtilities.checkForIdenticalInstances(infCatalystInst, null);
 				inferredCatalyst.put(catalystInst, infCatalystInst);
-				infReactionInst.addAttributeValue(catalystActivity, infCatalystInst);
-			} 
-		} 
+			} else {
+				logger.info("Inferred catalyst already exists");
+			}
+			infReactionInst.addAttributeValue(catalystActivity, inferredCatalyst.get(catalystInst));
+		}
+		logger.info("Completed catalyst inference");
 		return true;
 	}
 	
 	@SuppressWarnings("unchecked")
 	// Function used to infer regulation instances. Logic existed for regulators that had CatalystActivity and Event instances, but they have never come up in the many times this has been run.
-	private static List<GKInstance> inferReactionRegulations(GKInstance reactionInst) throws InvalidAttributeException, Exception
+	private static List<GKInstance> inferReactionRegulations(GKInstance reactionInst) throws Exception
 	{
-		List<GKInstance> inferredRegulations = new ArrayList<GKInstance>();
-		for (GKInstance regulatedInst : (Collection<GKInstance>) reactionInst.getAttributeValuesList("regulatedBy"))
+		List<GKInstance> inferredRegulations = new ArrayList<>();
+		Collection<GKInstance> regulationInstances = (Collection<GKInstance>) reactionInst.getAttributeValuesList("regulatedBy");
+		logger.info("Total RegulatedBy instances: " + regulationInstances.size());
+		if (regulationInstances.size() > 0) {
+			logger.info("Regulation instances: " + regulationInstances);
+		}
+		for (GKInstance regulationInst : regulationInstances)
```

Comment:
Again, there is an if-statement checking that the iterated structure has elements, so the loop on that iterated structure (`regulationInstances` in this case) probably should be moved to be within that if-statement.